storagenode/pieces/readwrite.go

```diff
@@ -113,14 +302,17 @@ func (r *Reader) Seek(offset int64, whence int) (int64, error) {
 
 // ReadAt reads data at the specified offset
 func (r *Reader) ReadAt(data []byte, offset int64) (int, error) {
+	if r.formatVersion >= storage.FormatV1 {
```

Comment:
Let's use a switch, that way we can detect the missing places which need the additional case.

Reply:
Same situation here. I'm not sure if we'll want multiple paths here, or if it will always just be `FormatV0` versus "everything else". Certainly if we end up needing more than 2 paths here, a switch will be the right way to go, but I don't want to start requiring that if it turns out 2 paths are all we ever need.